zerver/views/__init__.py

```diff
@@ -728,16 +728,22 @@ def send_registration_completion_email(email, request, realm_creation=False):
 not be the member of any current realm. The realm is created with domain same as the that of the user's email.
 When there is no unique_open_realm user registrations are made by visiting /register/domain_of_the_realm.
 """
-def create_realm(request):
-    # type: (HttpRequest) -> HttpResponse
-    if not settings.OPEN_REALM_CREATION:
-        return render_to_response("zerver/realm_creation_disabled.html")
+def create_realm(request, creation_key=None):
+    # type: (HttpRequest, str) -> HttpResponse
+    if not settings.OPEN_REALM_CREATION and creation_key is None:
+        return render_to_response("zerver/realm_creation_failed.html", {'message': 'New organization creation disabled.'})
+    if not settings.OPEN_REALM_CREATION and creation_key is not None:
+        if (not RealmCreationmKey.objects.filter(creation_key=creation_key).exists()):
+            return render_to_response("zerver/realm_creation_failed.html", {'message': 'The organization creation link has been expired or is not valid.'})
+            RealmCreationmKey.objects.get(creation_key=creation_key).delete()
```

Comment:
Also, I think this overall block would be easier to read if you had an outer `if not settings.OPEN_REALM_CREATION` and then did the rest of the `if` statements in a block inside there.